ThunderTable/InputDatePickerRow.swift

```diff
@@ -11,9 +11,73 @@ import UIKit
 /// A row which displays a date picker in the keyboard for the user
 /// to select a date and formats the date nicely
 open class InputDatePickerRow: InputTableRow {
+    
+    /// An direct map enum to `UIDatePickerStyle` so we can provide the `preferredDatePickerStyle` on iOS
+    /// versions before iOS 13.4 without getting "Stored properties cannot be marked potentially unavailable with '@available'"
+    /// compiler warning!
+    public enum PickerStyle: Int {
+        /// Automatically pick the best style available for the current platform & mode.
+        case automatic = 0
+
+        /// Use the wheels (UIPickerView) style. Editing occurs inline.
+        case wheels = 1
+
+        /// Use a compact style for the date picker. Editing occurs in an overlay.
+        case compact = 2
+
+        /// Use a style for the date picker that allows editing in place.
+        case inline = 3
+        
+        @available (iOS 13.4, *)
+        var datePickerStyle: UIDatePickerStyle {
+            switch self {
+            case .automatic:
+                return .automatic
+            case .compact:
+                return .compact
+            case .inline:
+                if #available(iOS 14.0, *) {
+                    return .inline
+                } else {
+                    return .automatic
+                }
+            case .wheels:
+                return .wheels
+            }
+        }
+        
+        /// Returns the correct cell class for the style given the mode the date picker is set to
+        /// - Parameter mode: The date picker mode the date picker is in
+        /// - Returns: A table view cell class used to render the cell
+        func cellClassFor(mode: UIDatePicker.Mode) -> UITableViewCell.Type {
+            
+            // Even old versions of iOS can support `inline` no matter how much of a hot mess it is visually!
+            if self == .inline {
```

Comment:
Do we need this here.
If iOS 14, then it would return this anyway below.
If not iOS 14 then it can never be `.inline`